plugins/module_utils/foreman_helper.py

```diff
@@ -809,29 +839,37 @@ def _copy_entity(self, resource, desired_entity, current_entity, params):
         return self.resource_action(resource, 'copy', payload)
 
     def _revert_entity(self, resource, current_entity, params):
-        """Revert a given entity
-
-            Parameters:
-                resource (string): Plural name of the api resource to manipulate
-                current_entity (dict): Current properties of the entity
-                params (dict): Lookup parameters (i.e. parent_id for nested entities) (optional)
-            Return value:
-                The new current state of the entity
+        """
+        Revert a given entity
+
+        :param resource: Plural name of the api resource to manipulate
+        :type resource: str
+        :param current_entity: Current properties of the entity
+        :type current_entity: dict
+        :param params: Lookup parameters (i.e. parent_id for nested entities)
+        :type params: dict, optional
+
+        :return: The new current state of the entity
+        :rtype: dict
         """
         payload = {'id': current_entity['id']}
         if params:
             payload.update(params)
         return self.resource_action(resource, 'revert', payload)
 
     def _delete_entity(self, resource, current_entity, params):
-        """Delete a given entity
-
-            Parameters:
-                resource (string): Plural name of the api resource to manipulate
-                current_entity (dict): Current properties of the entity
-                params (dict): Lookup parameters (i.e. parent_id for nested entities) (optional)
-            Return value:
-                The new current state of the entity
+        """
+        Delete a given entity
+
+        :param resource: Plural name of the api resource to manipulate
+        :type resource: str
+        :param current_entity: Current properties of the entity
+        :type current_entity: dict
+        :param params: Lookup parameters (i.e. parent_id for nested entities)
+        :type params: dict, optional
+
+        :return: The new current state of the entity
+        :rtype: dict
```

Comment:
this is IMHO wrong, the method returns `None`

Reply:
With `None` being "the new state of the entity", so it is technically `NoneType`, but we always meant `Union(NoneType, dict)`.

Reply:
So that's also what https://github.com/sphinx-doc/sphinx/issues/1874 recommends and I've implemented it now ;)